py/desitarget/mock/selection.py

```diff
@@ -362,6 +362,8 @@ def density_select(self, targets, truth, source_name, target_name, density=None,
             if density and n_in_brick > 0:
                 mock_density = n_in_brick / brick_area
                 desired_density = float(self.brick_info['FLUC_EBV_{}'.format(source_name)][brickindx] * density)
+                if desired_density < 0.0:
+                    desired_density = 0.0
 
```

Comment:
Why would `desired_density` ever be negative?  It seems to me that this edge case should be captured much earlier..

Reply:
I agree this should be captured earlier. I believe that `desired_densities` are being generated with a normal distribution around the mean desired density, that might be the source of the problem. But in the way density_select is written we weren't protected against that case.